ddtrace/settings/config.py

```diff
@@ -328,10 +328,11 @@ def __init__(self):
 
         self.http = HttpConfig(header_tags=self.trace_http_header_tags)
         self._tracing_enabled = asbool(os.getenv("DD_TRACE_ENABLED", default=True))
-        self._remote_config_enabled = asbool(os.getenv("DD_REMOTE_CONFIGURATION_ENABLED", default=True))
+        self._remote_config_enabled = asbool(os.getenv("_DD_REMOTE_CONFIGURATION_ENABLED", default=True))
```

Comment:
Are we allowed to just make this internal now? We should probably at least have a release note for customers as I know some rely on RCM being disabled with their current configuration.

Reply:
Good point. With the regression fixed we don't ever expect users to need to disable RCM. Besides this changes also makes the RCM worker start-up lazy: if there are no products registered with RC then the workers won't start. If the env var is set, it would effectively become a no-op.

Reply:
Spoke with team, can we deprecate `DD_REMOTE_CONFIGURATION_ENABLED` and have that be a fall back, that way for users that are disabling it, they won't all of the sudden crash again.

Reply:
This PR follows the fix for the gevent regression #7649, and besides makes RC lazy, meaning that if there are no products requiring it, the RC workers won't start. Users not relying on RC won't have it started anyway so I don't expect any impact in this regard.

Reply:
RC is not something that can be used directly, it's more an internal dependencies for other products to use. In this sense, I think it makes sense for RC to be enabled "on-demand" via other products (e.g. a user requesting DI, ASM etc...)